crawlers/mooncrawl/mooncrawl/cu_reports_crawler/cli.py

```diff
@@ -0,0 +1,616 @@
+import argparse
+import datetime
+from moonstream.client import Moonstream  # type: ignore
+import time
+import requests
+import json
+
+from typing import Any, Dict, Union
+
+from uuid import UUID
+
+from .queries import tokenomics_queries, cu_bank_queries
+
+from ..settings import CUSTOM_CRAWLER_S3_BUCKET, CUSTOM_CRAWLER_S3_BUCKET_PREFIX
+
+
+def recive_S3_data_from_query(
+    client: Moonstream,
+    token: Union[str, UUID],
+    query_name: str,
+    params: Dict[str, Any],
+    time_await: int = 2,
+    max_retries: int = 20,
+) -> Any:
+
+    """
+    Await the query to be update data on S3 with if_modified_since and return new the data.
+    """
+
+    keep_going = True
+
+    repeat = 0
+
+    if_modified_since_datetime = datetime.datetime.utcnow()
+    if_modified_since = if_modified_since_datetime.strftime("%a, %d %b %Y %H:%M:%S GMT")
+
+    time.sleep(2)
+
+    data_url = client.exec_query(
+        token=token,
+        name=query_name,
+        params=params,
+    )  # S3 presign_url
+
+    while keep_going:
+        time.sleep(time_await)
+        try:
+            data_response = requests.get(
+                data_url.url,
+                headers={"If-Modified-Since": if_modified_since},
+                timeout=5,
+            )
+        except Exception as e:
+            print(e)
+            continue
+
+        if data_response.status_code == 200:
+            break
+
+        repeat += 1
+
+        if repeat == max_retries // 2:
+            data_url = client.exec_query(
+                token=token,
+                name=query_name,
+                params=params,
+            )
+
+        if repeat > max_retries:
+            print("Too many retries")
+            break
+    return data_response.json()
+
+
+def generate_report(
+    client: Moonstream,
+    token: Union[str, UUID],
+    query_name: str,
+    params: Dict[str, Any],
+    bucket_prefix: str,
+    bucket: str,
+    key: str,
+):
+    """
+    Generate the report.
+    """
+
+    try:
+
+        json_data = recive_S3_data_from_query(
+            client=client,
+            token=token,
+            query_name=query_name,
+            params=params,
+        )
+
+        client.upload_query_results(
+            json.dumps(json_data),
+            bucket,
+            f"{bucket_prefix}/{key}",
+        )
+        print(f"https://{bucket}/{bucket_prefix}/{key}")
```

Comment:
Can we have here logging like:
`Report generated and results uploaded at: https;//...`